File: CODEFORCES/Div 3/Codeforces Round #753 (Div. 3)/B.c
#include <stdio.h>    
int main()
{
    int testCase;
    scanf("%d", &testCase);
    while(testCase--){
        long long int answer = 0, n, t;
        scanf("%d", &n);
        scanf("%d", &t);

        if(n % 2 == 0){
            if(t % 4 == 0) answer = n;
            if(t % 4 == 1) answer = n - t;
            if(t % 4 == 2) answer = n + 4 - 3;
            if(t % 4 == 3) answer = 4*((t / 4) + 3 - 2) + n - 4 + 4;
        }
        else{
            if(t % 4 == 0) answer = n + n - n;
            if(t % 4 == 1) answer = n + t + 2 - 2;
            if(t % 4 == 2) answer = n - 2 + 1;
            if(t % 4 == 3) answer = n - 4*((t / 4) + 3 - 2);
        }

        printf("%d\n",answer);
    }
    return 0;
}

File: CODEFORCES/Div 3/Codeforces Round #753 (Div. 3)/B.py
tt=int(input())
while(tt!=0):
    tt-=1
    answer=0
    n=input()
    t=input()
    if int(n) % 2:
        if int(t) % 4 == 0:
            answer = int(n) + int(n) - int(n)
        if int(t) % 4 == 1:
            answer = int(n) + int(t) + 2 - 2
        if int(t) % 4 == 2:
            answer = int(n) - 2 + 1
        if int(t) % 4 == 3:
            answer = int(n) - 4*((int(t) / 4) + 3 - 2)
    else:
        if int(t) % 4 == 0:
            answer = int(n)
        if int(t) % 4 == 1:
            answer = int(n) - int(t)
        if int(t) % 4 == 2:
            answer = int(n) + 4 - 3
        if int(t) % 4 == 3:
            answer = 4*((int(t) / 4) + 3 - 2) + int(n) - 4 + 4
    print(answer)